tests/console/validate_lvm_with_raid1.pm

```diff
@@ -0,0 +1,89 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Simple LVM partition validation
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.com>
+
+use strict;
+use warnings;
+use base "y2_module_consoletest";
+use testapi;
+use utils;
+use Test::Assert ':all';
+use scheduler 'get_test_data';
+
+my $config;
+my @expectedDevs;
+my $expected_num_devs;
+
+sub run {
+    my $self = shift;
+
+    select_console 'root-console';
+
+    $config            = get_test_data();
+    @expectedDevs      = (keys %{$config->{disks}});
+    $expected_num_devs = scalar @expectedDevs;
+
+    _lvm_partitioning();
+    _raid1_partitioning();
+    $self->_raid_can_boot_with_deleted_hdd();
```

Comment:
you can organize this in different way, as it looks strange that only one function have the `$self` when they all look the same, just because of `$self->wait_boot;`. For example you can have a `_check_raid1_partitioning ` `remove_disk`, `check_removed_disk` `_reboot`, and finally `_check_raid1_partitioning` and you pass this one a different parameter for the expected num of disks. `_reboot` can include:
```
record_info('system reboots');
type_string "shutdown -r now\n";
$self->wait_boot;
select_console 'root-console';
```
So it will be good to separate this logic in case this part doesn't work in other system, like Rodion mentioned in another comment and you need some adjustments and also it looks to me a different function so pass the $self might make more sense.

Reply:
i tried to address this but it is not validated that it works